subsys/shell/modules/bt.c

```diff
@@ -0,0 +1,2782 @@
+	.passkey_display = NULL,
+	.passkey_entry = auth_passkey_entry,
+	.passkey_confirm = NULL,
+#if defined(CONFIG_BLUETOOTH_BREDR)
+	.pincode_entry = auth_pincode_entry,
+#endif
+	.cancel = auth_cancel,
+	.pairing_confirm = auth_pairing_confirm,
+};
+
+static struct bt_conn_auth_cb auth_cb_all = {
+	.passkey_display = auth_passkey_display,
+	.passkey_entry = auth_passkey_entry,
+	.passkey_confirm = auth_passkey_confirm,
+#if defined(CONFIG_BLUETOOTH_BREDR)
+	.pincode_entry = auth_pincode_entry,
+#endif
+	.cancel = auth_cancel,
+	.pairing_confirm = auth_pairing_confirm,
+};
+
+static int cmd_auth(int argc, char *argv[])
+{
+	if (argc < 2) {
+		return -EINVAL;
+	}
+
+	if (!strcmp(argv[1], "all")) {
+		bt_conn_auth_cb_register(&auth_cb_all);
+	} else if (!strcmp(argv[1], "input")) {
+		bt_conn_auth_cb_register(&auth_cb_input);
+	} else if (!strcmp(argv[1], "display")) {
+		bt_conn_auth_cb_register(&auth_cb_display);
+	} else if (!strcmp(argv[1], "yesno")) {
+		bt_conn_auth_cb_register(&auth_cb_display_yes_no);
+	} else if (!strcmp(argv[1], "none")) {
+		bt_conn_auth_cb_register(NULL);
+	} else {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int cmd_auth_cancel(int argc, char *argv[])
+{
+	struct bt_conn *conn;
+
+	if (default_conn) {
+		conn = default_conn;
+	} else if (pairing_conn) {
+		conn = pairing_conn;
+	} else {
+		conn = NULL;
+	}
+
+	if (!conn) {
+		printk("Not connected\n");
+		return 0;
+	}
+
+	bt_conn_auth_cancel(conn);
+
+	return 0;
+}
+
+static int cmd_auth_passkey_confirm(int argc, char *argv[])
+{
+	if (!default_conn) {
+		printk("Not connected\n");
+		return 0;
+	}
+
+	bt_conn_auth_passkey_confirm(default_conn);
+
+	return 0;
+}
+
+static int cmd_auth_pairing_confirm(int argc, char *argv[])
+{
+	if (!default_conn) {
+		printk("Not connected\n");
+		return 0;
+	}
+
+	bt_conn_auth_pairing_confirm(default_conn);
+
+	return 0;
+}
+
+static int cmd_auth_passkey(int argc, char *argv[])
+{
+	unsigned int passkey;
+
+	if (!default_conn) {
+		printk("Not connected\n");
+		return 0;
+	}
+
+	if (argc < 2) {
+		return -EINVAL;
+	}
+
+	passkey = atoi(argv[1]);
+	if (passkey > 999999) {
+		printk("Passkey should be between 0-999999\n");
+		return 0;
+	}
+
+	bt_conn_auth_passkey_entry(default_conn, passkey);
+
+	return 0;
+}
+#endif /* CONFIG_BLUETOOTH_SMP) || CONFIG_BLUETOOTH_BREDR */
+
+#if defined(CONFIG_BLUETOOTH_BREDR)
+static int cmd_auth_pincode(int argc, char *argv[])
+{
+	struct bt_conn *conn;
+	u8_t max = 16;
+
+	if (default_conn) {
+		conn = default_conn;
+	} else if (pairing_conn) {
+		conn = pairing_conn;
+	} else {
+		conn = NULL;
+	}
+
+	if (!conn) {
+		printk("Not connected\n");
+		return 0;
+	}
+
+	if (argc < 2) {
+		return -EINVAL;
+	}
+
+	if (strlen(argv[1]) > max) {
+		printk("PIN code value invalid - enter max %u digits\n", max);
+		return 0;
+	}
+
+	printk("PIN code \"%s\" applied\n", argv[1]);
+
+	bt_conn_auth_pincode_entry(conn, argv[1]);
+
+	return 0;
+}
+
+static int str2bt_addr(const char *str, bt_addr_t *addr)
```

Comment:
Except for the type, but I guess we can make it str2bt_addr_le call str2bt_addr and just deal with type.